### PR TITLE
Add Stripe Adapter and IBraintreeGateway to DI 

### PR DESCRIPTION
**Pull request by Hinton**
## Objective
In an effort to make the `StripePaymentService` testable, I've refactored the third party logic to be injected as dependencies which can be mocked during testing.

To achive this I had to add a wrapper to stripe, and add a new DI provider for IBraintreeGateway.

### CODE REVIEW

**Review by MGibson1 — DISMISSED** [review]
Nice!

File: src/Core/Utilities/ServiceCollectionExtensions.cs [hunk 1]
```diff
@@ -184,6 +184,18 @@ public static void AddDefaultServices(this IServiceCollection services, GlobalSe
             // Required for UserService
             services.AddWebAuthn(globalSettings);
 
+            services.AddSingleton<IStripeAdapter, StripeAdapter>();
+            services.AddSingleton<Braintree.IBraintreeGateway>((serviceProvider) =>
```
Comment (justindbaur) [inline src/Core/Utilities/ServiceCollectionExtensions.cs:188]: I had discussed the possibility of doing this on #1454. I think this is a really good idea!

**Review by MGibson1 — APPROVED** [review]

**Review by MGibson1 — CHANGES_REQUESTED** [review]
Unfortunately `SubscriptionUpdate` uses Stripe. We should probably make it use the adapter.

To me that also means that a static factory is preferrable. I'll leave up to you, though

**Comment by Hinton** [comment]
@MGibson1 `SubscriptionUpdate` only extends the business data model right? Which doesn't need to be mocked.

**Comment by MGibson1** [comment]
yeah, you're right, there are no services in there.

**Pull request merged**